pkg/ddc/thin/engine.go

```diff
@@ -96,3 +105,9 @@ func Precheck(client client.Client, key types.NamespacedName) (found bool, err e
 	var obj datav1alpha1.ThinRuntime
 	return utils.CheckObject(client, key, &obj)
 }
+
+// isVirtualDatasetRuntime judge if this runtime is used for handling dataset mounting another dataset.
+func isVirtualDatasetRuntime(runtime *datav1alpha1.ThinRuntime) bool {
+	profileName := runtime.Spec.ThinRuntimeProfileName
+	return profileName == ""
```

Comment:
If Dataset of ThinRuntime used existed PVC as mountpoint, its profile will be null too. But this situation is not virtual dataset. `isVirtualDatasetRuntime` could not use profile is null as condition. I think it's better to use dataset.mountpoint is `dataset://` as condition instead.